### ISSUE TITLE
Unable to boot patched.x230.img

### ISSUE THREAD

**Issue by craftyguy** [hamishcoleman/thinkpad-ec#78]
I tried to give this a go, after enabling the keyboard and battery patches, but when booting from the USB disk that was flashed with patched.x230.img, I only see a screen that says `GRUB` and nothing else. I've tried re-generating the img file, several USB drives, and two different USB ports on the x230.

Build details below!

```

Your build has completed with the following details:

Buildinfo: v1-229-ged6ea6-dirty (20180517) patched.x230.img
Built: d1ee650c9312c91f124b85c423b7eecf0af654ef  x230.G2HT35WW.s01D3000.FL2

Patches applied:
x230.G2HT35WW.img.d/001_keysym.patch x230.G2HT35WW.img.d/002_dead_keys.patch
x230.G2HT35WW.img.d/003_keysym_replacements.patch
x230.G2HT35WW.img.d/004_fn_keys.patch x230.G2HT35WW.img.d/005_fn_key_swap.patch
x230.G2HT35WW.img.d/006_battery_validate.patch

Description: x230 BIOS 2.72 (G2ETB2WW) EC 1.14 (G2HT35WW)
./scripts/geteltorito -o patched.x230.img.tmp patched.x230.iso
Booting catalog starts at sector: 20
Manufacturer of CD: NERO BURNING ROM
Image architecture: x86
Boot media type is: harddisk
El Torito image starts at sector 27 and has 65536 sector(s) of 512 Bytes

Image has been written to file "patched.x230.img.tmp".
./scripts/fix_mbr patched.x230.img.tmp
mv patched.x230.img.tmp patched.x230.img


Your build has completed with the following details:

Buildinfo: v1-229-ged6ea6-dirty (20180517) patched.x230.img
Built: d1ee650c9312c91f124b85c423b7eecf0af654ef  x230.G2HT35WW.s01D3000.FL2

Patches applied:
x230.G2HT35WW.img.d/001_keysym.patch x230.G2HT35WW.img.d/002_dead_keys.patch
x230.G2HT35WW.img.d/003_keysym_replacements.patch
x230.G2HT35WW.img.d/004_fn_keys.patch x230.G2HT35WW.img.d/005_fn_key_swap.patch
x230.G2HT35WW.img.d/006_battery_validate.patch

Description: x230 BIOS 2.72 (G2ETB2WW) EC 1.14 (G2HT35WW)
```

**Comment by craftyguy** [comment]
Some more information:

- EFI boot is disabled, booting legacy-only

-  BIOS version on system: 2.67

- EC version on system: 1.14

I don't think the versions have anything to do with it since it seems like I'm not even able to successfully boot the generated image to begin the process of flashing.

**Comment by craftyguy** [comment]
Well, if I enable EFI-only boot, I am able to successfully boot the image and apply the update. This is in contrast to the recommendation in the README that folks must use legacy boot. :man_shrugging:

**Comment by craftyguy** [comment]
Alright, epic fail on my part.

I suspect a disk in my boot order was giving the `GRUB` message and hanging the boot.

I removed the whole project, and rebuilt it. After correcting the boot order to only boot the usb drive with this EC update on it, I was able to apply is successfully for real. I did have to power off the unit and power it back on before the EC would flash. Simply soft rebooting the system didn't work. Sorry for the noise!
